pkg/apis/workspaces/v1alpha2/keyed.go

```diff
@@ -1,108 +1,11 @@
 package v1alpha2
 
 type Keyed interface {
-	Key() (string, error)
+	Key() string
 }
 
-func (union Component) Key() (string, error) {
-	key := ""
-	err := union.Visit(ComponentVisitor{
-		Container: func(container *ContainerComponent) error {
-			key = container.Name
-			return nil
-		},
-		Plugin: func(plugin *PluginComponent) error {
-			if plugin.Name != "" {
-				key = plugin.Name
-				return nil
-			}
-			return plugin.ImportReference.ImportReferenceUnion.Visit(ImportReferenceUnionVisitor{
-				Uri: func(uri string) error {
-					key = uri
-					return nil
-				},
-				Id: func(id string) error {
-					key = plugin.Id
-					if plugin.RegistryUrl != "" {
-						key = plugin.RegistryUrl + "/" + key
-					}
-					return nil
-				},
-				Kubernetes: func(cr *KubernetesCustomResourceImportReference) error {
-					key = cr.Name
-					if cr.Namespace != "" {
-						key = cr.Namespace + "/" + key
-					}
-					return nil
-				},
-			})
-		},
-		Kubernetes: func(k8s *KubernetesComponent) error {
-			key = k8s.Name
-			return nil
-		},
-		Openshift: func(os *OpenshiftComponent) error {
-			key = os.Name
-			return nil
-		},
-		Volume: func(vol *VolumeComponent) error {
-			key = vol.Name
-			return nil
-		},
-	})
-	return key, err
-}
-
-func (union PluginComponentsOverride) Key() (string, error) {
-	key := ""
-	err := union.Visit(PluginComponentsOverrideVisitor{
-		Container: func(container *ContainerComponent) error {
-			key = container.Name
-			return nil
-		},
-		Kubernetes: func(k8s *KubernetesComponent) error {
-			key = k8s.Name
-			return nil
-		},
-		Openshift: func(os *OpenshiftComponent) error {
-			key = os.Name
-			return nil
-		},
-		Volume: func(vol *VolumeComponent) error {
-			key = vol.Name
-			return nil
-		},
-	})
-	return key, err
-}
+type TopLevelLists map[string][]string
```

Comment:
In this commit https://github.com/devfile/api/pull/122/commits/3c18672f7de1f419ae53b732b16804c0b38b284c, I added documentation and also use a new `KeyedList` type instead of `[]string` as the type of map items.

Reply:
Do those docs make things clearer ?

Reply:
Yeah that helps, thanks.